packages/ui5-tooling-transpile/lib/task.js

```diff
@@ -119,6 +119,24 @@ module.exports = async function ({ workspace /*, dependencies*/, taskUtil, optio
 					//traceResolution: true,
 				};
 
+				// update the sources map to declare the modules with the full module name
+				for await (const resourcePath of Object.keys(sourcesMap)) {
+					const source = sourcesMap[resourcePath];
+					let moduleName = /^\/resources\/(.*)\.ts$/.exec(resourcePath)?.[1];
+					if (moduleName?.endsWith(".gen.d")) {
+						moduleName = /^(.*)\.gen\.d$/.exec(moduleName)?.[1] || moduleName;
```

Comment:
Ummm... isn't the OR part in the end of this line (and the question mark) unneeded? You already checked that the module name ends with ".gen.d".

Reply:
Relict from previous logic - will adapt